### PR TITLE
DOC: partial standardization and validation of scipy.stats reference guide

### PR DESCRIPTION
**Pull request by brandondavid**
Minor corrections and reformatting.  
Standardization of docstyle for string parameters.  
Partial enforcement of [PEP 257](https://www.python.org/dev/peps/pep-0257/).
Partial enforcement of the [numpydoc docstring guide](https://numpydoc.readthedocs.io/en/latest/format.html).

### CODE REVIEW

File: scipy/stats/stats.py [hunk 1]
```diff
@@ -4732,13 +4788,13 @@ def kstest(rvs, cdf, args=(), N=20, alternative='two-sided', mode='approx'):
...
     random variable against a given distribution G(x). Under the null
-    hypothesis the two distributions are identical, F(x)=G(x). The
+    hypothesis, the two distributions are identical, F(x)=G(x). The
     alternative hypothesis can be either 'two-sided' (default), 'less'
     or 'greater'. The KS test is only valid for continuous distributions.
 
     Parameters
     ----------
-    rvs : str, array or callable
+    rvs : str, array, or callable
```
Comment (rgommers) [inline scipy/stats/stats.py]: note, `array` should be either `ndarray` or `array-like`

File: scipy/stats/stats.py [hunk 2]
```diff
@@ -5440,25 +5505,28 @@ def ks_2samp(data1, data2, alternative='two-sided', mode='auto'):
...
     mode : {'auto', 'exact', 'asymp'}, optional
         Defines the method used for calculating the p-value.
-        Default is 'auto'.
+        The following options are available (default is 'auto'):
 
-        - 'exact' : use approximation to exact distribution of test statistic
-        - 'asymp' : use asymptotic distribution of test statistic
-        - 'auto' : use 'exact' for small size arrays, 'asymp' for large.
+		  * 'auto' : use 'exact' for small size arrays, 'asymp' for large
+		  * 'exact' : use approximation to exact distribution of test statistic
```
Comment (rgommers) [inline scipy/stats/stats.py]: indentation is off here. this causes:
```
13      E101 indentation contains mixed spaces and tabs
7       W191 indentation contains tabs
```
as seen in CI.

**Review by rgommers — APPROVED** [review]
Overall this looks good, just a few minor comments. @brandondavid you may want to configure your edits to always convert tabs to 4 spaces.

File: scipy/stats/morestats.py [hunk 3]
```diff
@@ -2374,9 +2381,13 @@ def binom_test(x, n=None, p=0.5, alternative='two-sided'):
...
         default value is p = 0.5
-    alternative : {'two-sided', 'greater', 'less'}, optional
-        Indicates the alternative hypothesis. The default value is
-        'two-sided'.
+    alternative : {'two-sided', 'less', 'greater'}, optional
+        Defines the alternative hypothesis.
+        The following options are available (default is 'two-sided'):
+
+          * 'two-sided'
+          * 'less': ones-sided
```
Comment (chrisb83) [inline scipy/stats/morestats.py]: typo

Not sure if the list helps without further information. One could add a comment about the null hypothesis and the alternative hypothesis

File: scipy/stats/stats.py [hunk 4]
```diff
@@ -4752,14 +4808,19 @@ def kstest(rvs, cdf, args=(), N=20, alternative='two-sided', mode='approx'):
...
-    alternative : {'two-sided', 'less','greater'}, optional
-        Defines the alternative hypothesis (see explanation above).
-        Default is 'two-sided'.
-    mode : 'approx' (default) or 'asymp', optional
+    alternative : {'two-sided', 'less', 'greater'}, optional
+        Defines the alternative hypothesis.
+        The following options are available (default is 'two-sided'):
+
+          * 'two-sided'
+          * 'less': ones-sided, see explanation in Notes
```
Comment (chrisb83) [inline scipy/stats/stats.py]: typo ones -> one

File: scipy/stats/stats.py [hunk 5]
```diff
@@ -6026,18 +6107,26 @@ def brunnermunzel(x, y, alternative="two-sided", distribution="t",
...
-    distribution: 't' or 'normal', optional
-        Whether to get the p-value by t-distribution or by standard normal
-        distribution.
-        Defaults value is 't' .
+    alternative : {'two-sided', 'less', 'greater'}, optional
+        Defines the alternative hypothesis.
+        The following options are available (default is 'two-sided'):
+
+          * 'two-sided'
+          * 'less': ones-sided
```
Comment (chrisb83) [inline scipy/stats/stats.py]: same

File: scipy/stats/stats.py [hunk 6]
```diff
@@ -6151,19 +6240,18 @@ def combine_pvalues(pvalues, method='fisher', weights=None):
...
+    method : {'fisher', 'pearson', 'tippett', 'stouffer', 'mudholkar_george'}, optional
+        Name of method to use to combine p-values.
+        The following methods are available (default is 'fisher'):
+
+          * 'fisher': Fisher's method (Fisher's combined probability test), the
+            sum of the logarithm of the p-values
+          * 'pearson': Pearson's method (similar to Fisher's but uses sum of the
+            complement of the p-values inside the logarithms)
+          * 'tippett': Tippett's method (minimum of p-values)
+          * 'touffer': Stouffer's Z-score method
```
Comment (chrisb83) [inline scipy/stats/stats.py]: stouffer

**Comment by rgommers** [comment]
@brandondavid this will need a merge with master or a rebase to make CircleCI build the docs, because there's a merge conflict:
```
From git://github.com/scipy/scipy
 * branch                refs/pull/10863/merge -> FETCH_HEAD
fatal: Not possible to fast-forward, aborting.
```

File: scipy/stats/stats.py [hunk 7]
```diff
@@ -2395,15 +2453,21 @@ def zmap(scores, compare, axis=0, ddof=0):
     >>> b = [0, 1, 2, 3, 4]
     >>> zmap(a, b)
     array([-1.06066017,  0.        ,  0.35355339,  0.70710678])
+    
     """
     scores, compare = map(np.asanyarray, [scores, compare])
-    mns = compare.mean(axis=axis, keepdims=True)
-    sstd = compare.std(axis=axis, ddof=ddof, keepdims=True)
-    return (scores - mns) / sstd
+    mns = compare.mean(axis=axis)
```
Comment (chrisb83) [inline scipy/stats/stats.py]: Did you change the code or is it from a merge? I think it is better to split code changes and doc changes

File: scipy/stats/stats.py [hunk 8]
```diff
@@ -3831,9 +3901,9 @@ def pointbiserialr(x, y):
     Returns
     -------
     correlation : float
-        R value
+        R value.
     pvalue : float
-        2-tailed p-value
+        2-tailed p-value.
```
Comment (chrisb83) [inline scipy/stats/stats.py]: Two-sided

File: scipy/stats/stats.py [hunk 9]
```diff
@@ -4288,24 +4362,27 @@ def ttest_1samp(a, popmean, axis=0, nan_policy='propagate'):
...
+          * 'omit': performs the calculations ignoring nan values
 
     Returns
     -------
     statistic : float or array
-        t-statistic
+        t-statistic.
     pvalue : float or array
-        two-tailed p-value
+        Two-tailed p-value.
```
Comment (chrisb83) [inline scipy/stats/stats.py]: Two-sided

File: scipy/stats/stats.py [hunk 10]
```diff
@@ -4654,20 +4733,23 @@ def ttest_rel(a, b, axis=0, nan_policy='propagate'):
...
```
Comment (chrisb83) [inline scipy/stats/stats.py]: Two-sided

**Review by chrisb83 — COMMENTED** [review]
Some minor comments

**Comment by rgommers** [comment]
This is all green and looks good, in it goes. Thanks @brandondavid, and thanks @chrisb83 for reviewing

**Pull request merged**